core/src/test/java/org/apache/iceberg/TestRewriteFiles.java

```diff
@@ -384,6 +386,116 @@ public void testRewriteDataAndAssignOldSequenceNumber() {
     assertThat(listManifestFiles()).hasSize(4);
   }
 
+  @TestTemplate
+  public void testRewriteDataAndAssignOldSequenceNumbersShouldNotDropDeleteFiles() {
```

Comment:
I've gone through this test and I think that the assertions are correct for the sequence of changes.